proto/cosmos/tx/v1beta1/service.proto

```diff
@@ -17,6 +20,63 @@ service Service {
     rpc GetTx(GetTxRequest) returns (GetTxResponse) {
         option (google.api.http).get = "/cosmos/tx/v1beta1/tx/{hash}";
     }
+
+    // TxsByEvents fetches txs by event.
+    rpc TxsByEvents(GetTxsEventRequest) returns (TxsByEventsResponse) {
+        option (google.api.http).get = "/cosmos/tx/v1beta1/txs";
+    }
+}
+
+// GetTxsEventRequest is the request type for the Service.TxsByEvents
+// RPC method.
+message GetTxsEventRequest {
+
+    // event is the transaction event type.
+    string event = 1;
+
+    // pagination defines an pagination for the request.
+    cosmos.base.query.v1beta1.PageRequest pagination = 2;
+
+}
+
+// TxsByEventsResponse is the response type for the Service.TxsByEvents
+// RPC method.
+message TxsByEventsResponse {
+    // txs is the list of queried transactions.
+    repeated TxRes txs = 1;
+
+    // pagination defines an pagination for the response.
+    cosmos.base.query.v1beta1.PageResponse pagination = 2;
+
+}
+
+// TxRes defines a structure containing relevant tx data and metadata.
+message TxRes {
```

Comment:
@alexanderbez maybe `TxServiceResponse`? Since it's specific to the TxService here.